[15.0][FIX] project_timeline: Allow to install this module with project_enterprise

alan196 opened the pull request:
This PR fixes an issue added in commit https://github.com/OCA/project/commit/3689f1dfccf755c5d862c0e779ab0364215dea3a

When the module project_timeline is installed in a database with the module project_enterprise, it raises the following error:

```
psycopg2.errors.DuplicateColumn: column "planned_date_end" specified more than once
```

This is because the field planned_date_end is defined in both modules.

This PR checks if the field is already defined in the query before adding it.

Fixes https://github.com/OCA/project/issues/1433

alan196 commented:
@HugoCLU @carlos-lopez-tecnativa Could you help me to review 🙏

pedrobaeza commented on project_timeline/report/project_report.py:
```diff
@@ -11,18 +11,33 @@ class ReportProjectTaskUser(models.Model):
 
     def _select(self):
+        res = super()._select()
+        if "planned_date_end" in res:
```
This is too broad. You may have other field `planned_date_end_custom` and do matching.

alan196 replied:
I already make the change @pedrobaeza could you help me to review if this is the change you requested 🙏

alan196 committed:
[FIX] project_timeline: Allow to install this module with project_enterprise
This PR fixes an issue added in commit

https://github.com/OCA/project/commit/3689f1dfccf755c5d862c0e779ab0364215dea3a

When the module project_timeline is installed in a database with the module
project_enterprise, it raises the following error:

psycopg2.errors.DuplicateColumn: column "planned_date_end" specified more than once

This is because the field planned_date_end is defined in both modules.

This PR change the field name from planned_date_end to planned_date_end_custom in order
to avoid the conflict.

pedrobaeza commented:
No, that was not what I requested. What I requested is that you make the `if` sentence more explicit, including quotes, comma or something, to make it match only when the exact column name is found, not in other case.

AaronHForgeFlow commented:
Superseded in https://github.com/OCA/project/pull/1564
